frontend/src/app/main/ui/viewer.cljs

```diff
@@ -60,6 +61,112 @@
+   [:div.viewer-bottom {:class (when left-bar "left-bar")}
+    [:div.reset {:on-click #(st/emit! dv/select-first-frame)} i/reset]
+    [:div.counter (str/join " / " [(+ index 1) num-frames])]
+    [:span]]])
+
+
+(mf/defc viewer-wrapper
+  [{:keys [wrapper-size scroll orig-frame orig-viewport-ref orig-size page file users current-viewport-ref
+           size frame interactions-mode overlays zoom close-overlay section index] :as props}]
+  (let [{clist :list} (mf/deref refs/comments-local)
+        show-comments-list (and (= section :comments) (= :show clist))]
+    [:*
+     [:& viewer-pagination {:index index :num-frames (count (:frames page)) :right-bar show-comments-list}]
+
+     (when show-comments-list
+       [:& comments-sidebar {:users users :frame frame :page page}])
+     
+     [:div.viewer-wrapper
+      {:style {:width (:width wrapper-size)
+               :height (:height wrapper-size)}}
+      [:& (mf/provider ctx/scroll-ctx) {:value @scroll}
+       [:div.viewer-clipper
+        [:*
+         (when orig-frame
+           [:div.viewport-container
+            {:ref orig-viewport-ref
+             :style {:width (:width orig-size)
+                     :height (:height orig-size)
+                     :position "relative"}}
+
+            [:& interactions/viewport
+             {:frame orig-frame
+              :base-frame orig-frame
+              :frame-offset (gpt/point 0 0)
+              :size orig-size
+              :page page
+              :file file
+              :users users
+              :interactions-mode :hide}]])
+
+         [:div.viewport-container
+          {:ref current-viewport-ref
+           :style {:width (:width size)
+                   :height (:height size)
+                   :position "relative"}}
+
+          [:& interactions/viewport
+           {:frame frame
+            :base-frame frame
+            :frame-offset (gpt/point 0 0)
+            :size size
+            :page page
+            :file file
+            :users users
+            :interactions-mode interactions-mode}]
+
+          (for [overlay overlays]
+            (let [size-over (calculate-size (:frame overlay) zoom)]
+              [:*
+               (when (or (:close-click-outside overlay)
+                         (:background-overlay  overlay))
+                 [:div.viewer-overlay-background
+                  {:class (dom/classnames
+                           :visible (:background-overlay overlay))
+                   :style {:width (:width wrapper-size)
+                           :height (:height wrapper-size)
+                           :position "absolute"
+                           :left 0
+                           :top 0}
+                   :on-click #(when (:close-click-outside overlay)
+                                (close-overlay (:frame overlay)))}])
+               [:div.viewport-container.viewer-overlay
+                {:id (str "overlay-" (str (:id (:frame overlay))))
```

Comment:
You can replace this with `(str "overlay-" (-> overlay :frame :id))` (removed the nested `str` call)